rust-src/crypto_common/src/types.rs

```diff
@@ -160,6 +161,26 @@ impl Deserial for OwnedContractName {
     }
 }
 
+impl Serial for Parameter<'_> {
+    fn serial<B: Buffer>(&self, out: &mut B) {
+        (self.0.len() as u16).serial(out);
+        crate::serial_vector_no_length(self.0, out)
+    }
+}
+
+impl Serial for OwnedParameter {
+    #[inline]
+    fn serial<B: Buffer>(&self, out: &mut B) { self.as_parameter().serial(out) }
+}
+
+impl Deserial for OwnedParameter {
+    fn deserial<R: ReadBytesExt>(source: &mut R) -> ParseResult<Self> {
+        let len: u16 = source.get()?;
```

Comment:
Perhaps it is worth explaining here why we do not check agains MAX_PARAMETER_SIZE